src/index.js

```diff
@@ -1,32 +1,62 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const XMLHttpRequest = require('xmlhttprequest').XMLHttpRequest;
+const API = 'https://rickandmortyapi.com/api/character/';
 
-var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
+function fetchAPI(url) {
+  const xhttp = new XMLHttpRequest();
+  return new Promise((resolve, reject) => {
+    xhttp.onreadystatechange = () => {
+      if (xhttp.readyState === 4) {
+        if (xhttp.status == 200) {
+          const data = JSON.parse(xhttp.responseText);
+          resolve(data);
+        } else {
+          return reject(xhttp.status);
```

Comment:
Te recomiendo ser mas claros en el reject, esto te permite identificar el error mas fácil. 

```suggestion
          reject(new Error("Algo salió mal."));
```